### PR TITLE
MTDSA-3844 Added controller logic for Amend loss claims

### PR DESCRIPTION
**Pull request by narendravijayarao**

### CODE REVIEW

File: test/v1/controllers/AmendLossClaimControllerSpec.scala [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+      auditService = mockAuditService,
+      cc = cc
+    )
+
+    MockedMtdIdLookupService.lookup(nino).returns(Future.successful(Right("test-mtd-id")))
+    MockedEnrolmentsAuthService.authoriseUser()
+  }
+
+  "amend" should {
+    "return a successful response with header X-CorrelationId and body" when {
```
Comment (MarkAKelly) [inline test/v1/controllers/AmendLossClaimControllerSpec.scala]: Doesn't check the existence of a correlation ID - will need adding.  I'll do the other tests under a support ticket.

File: app/v1/controllers/AmendLossClaimController.scala [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    }
+
+  private def processError(errorWrapper: ErrorWrapper) = {
+    errorWrapper.error match {
+      case BadRequestError
+           | NinoFormatError
+           | RuleIncorrectOrEmptyBodyError
+           | ClaimIdFormatError
+           | TypeOfClaimFormatError => BadRequest(Json.toJson(errorWrapper))
+      case RuleClaimTypeNotChanged | RuleTypeOfClaimInvalid => Forbidden(Json.toJson(errorWrapper))
```
[the remark this answers is not in the document]
Comment (narendravijayarao) [reply app/v1/controllers/AmendLossClaimController.scala]: Screenshot 2019-07-25 at 08 40 29

[the remark this answers is not in the document]
Comment (narendravijayarao) [reply app/v1/controllers/AmendLossClaimController.scala]: Screenshot 2019-07-25 at 08 46 17

File: test/v1/controllers/AmendLossClaimControllerSpec.scala (hunk 1 above)
Comment (narendravijayarao) [reply test/v1/controllers/AmendLossClaimControllerSpec.scala]: I added for all the tests

File: it/v1/endpoints/AmendLossClaimControllerISpec.scala [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+          val response: WSResponse = await(request().post(requestJson))
+          response.status shouldBe expectedStatus
+          response.json shouldBe Json.toJson(expectedBody)
+        }
+      }
+
+      serviceErrorTest(Status.BAD_REQUEST, "INVALID_TAXABLE_ENTITY_ID", Status.BAD_REQUEST, NinoFormatError)
+      serviceErrorTest(Status.BAD_REQUEST, "INVALID_CLAIM_ID", Status.BAD_REQUEST, ClaimIdFormatError)
+      serviceErrorTest(Status.FORBIDDEN, "INVALID_CLAIM_TYPE", Status.FORBIDDEN, RuleTypeOfClaimInvalid)
```
[the remark this answers is not in the document]
Comment (narendravijayarao) [reply it/v1/endpoints/AmendLossClaimControllerISpec.scala]: It is Forbidden and updated in the requirement spec.

[the remark this answers is not in the document]
Comment (MarkAKelly) [reply it/v1/endpoints/AmendLossClaimControllerISpec.scala]: :+1:

**Pull request merged**